processor/src/main/java/org/mapstruct/ap/internal/model/common/ParameterBinding.java

```diff
@@ -0,0 +1,124 @@
+/**
+ *  Copyright 2012-2016 Gunnar Morling (http://www.gunnarmorling.de/)
+ *  and/or other contributors as indicated by the @authors tag. See the
+ *  copyright.txt file in the distribution for a full listing of all
+ *  contributors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package org.mapstruct.ap.internal.model.common;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * Represents how one parameter of a method to be called is populated.
+ *
+ * @author Andreas Gudian
+ */
+public class ParameterBinding {
+
+    private final Type type;
+    private final String variableName;
+    private final boolean targetType;
+    private final boolean mappingTarget;
+
+    private ParameterBinding(Type parameterType, String variableName, boolean mappingTarget, boolean targetType) {
+        this.type = parameterType;
+        this.variableName = variableName;
+        this.targetType = targetType;
+        this.mappingTarget = mappingTarget;
+    }
+
+    /**
+     * @return the name of the variable (or parameter) that is being used as argument for the parameter being bound.
+     */
+    public String getVariableName() {
```

Comment:
But (and perhaps we should tackle this later).. If a localvar binds to a 'method to be called parameter', I would somehow expect that relation still to be there. Now the relation is only maintained because of the order in the list. Perhaps the `ParameterBinding` should wrap the original `Parameter`? Alternatively, we could maintain that relation where we use (e.g. method reference) in a map. Or, we should maintain the argument number somehow hard in the `Parameter` and the `ParameterBinding`.  I know there's no hard argument to do so, but I've a nagging feeling we should relate Parameter harder to ParameterBinding.